llvm/lib/Target/RISCV/RISCVTargetMachine.cpp

```diff
@@ -567,6 +568,7 @@ void RISCVPassConfig::addPreEmitPass() {
     addPass(createMachineCopyPropagationPass(true));
   addPass(&BranchRelaxationPassID);
   addPass(createRISCVMakeCompressibleOptPass());
+  addPass(createRISCVLatePeepholeOptPass());
```

Comment:
I'm not sure we should mess with branches after BranchRelaxationPass

Reply:
I'm checking my notes here and the pattern that generates the branch, i.e.:
```
bb.2:
  renamable $x10 = XORI killed renamable $x10, -1
  renamable $x10 = BEXTI killed renamable $x10, 13
  BNE killed renamable $x10, $x0, %bb.5
```
Seems to be formed by `MachineBlockPlacementPass`, so maybe we can move this change to the beginning of `addPreEmitPass` and still be able to optimize it.

Reply:
I would put it immediately before branch relaxation, unless you're also sure that machine copy propagation won't also introduce these kinds of branches.

Reply:
@lenary, MCP is definitely introducing these kinds of branches.